inventory.py

```diff
@@ -0,0 +1,143 @@
+        if self.proxy:
+            self.proxy += '+'
+        else:
+            self.proxy = ''
+            
+        # Get inventory file name from ENV VAR.
+        self.inventory_file = os.getenv('AI_INVENTORY')
+        if self.inventory_file:
+            self.inventory_path = os.path.dirname(os.path.realpath(__file__)) + '/' + self.inventory_file
+        else:
+            self.inventory_path = os.path.dirname(os.path.realpath(__file__)) + '/inventory.ini'
+        if not (os.path.isfile(self.inventory_path) and os.access(self.inventory_path, os.R_OK)):
+            print 'FATAL: The static inventory file ' + self.inventory_path + ' is either missing or not readable: Check path and permissions.'
+            print '       You may need to export the AI_INVENTORY environment variable to point to a static inventory file in the same dir as where '
+            print '           ' + os.path.realpath(__file__)
+            print '       is located.'
+            sys.exit()
+
+        # Read settings and parse CLI arguments.
+        self.read_inventory_template()
+        self.parse_cli_args()
+
+        data_to_print = ""
+        data_to_print += self.dict_to_json(self.inventory, True)
+        print(data_to_print)
+
+    #
+    # Read the inventory details from an Ansible inventory file
+    # * named 'inventory',
+    # * in *.ini format and
+    # located in the same place as this script.
+    #
+    def read_inventory_template(self):
+
+        _config = ConfigParser.SafeConfigParser(allow_no_value=True)
+        _config.optionxform = self.prepend_proxy
+        
+        _config.read(os.path.dirname(os.path.realpath(__file__)) + '/' + self.inventory_file)
+
+        for _section in _config.sections():
+            for (_key, _value) in _config.items(_section):
+                self.push(self.inventory, _section, _key)
+
+    #
+    # Prepends proxy before host.
+    #
+    def prepend_proxy(self, _string):
+        #return re.sub('AI_PROXY\+', self.proxy, _string)
+        return re.sub('^', self.proxy, _string)
+
+
+    #
+    # Process command line arguments.
+    #
+    def parse_cli_args(self):
+        parser = argparse.ArgumentParser(description='Produce an Ansible Inventory file.')
+        parser.add_argument('--list', action='store_true', default=True,
+                            help='List instances (default: True)')
+        self.args = parser.parse_args()
+
+    #
+    # Push an element into an array that may or may not have been defined already in the dict.
+    #
+    def push(self, _dict, _key, _element):
+        if _key in _dict:
+            _dict[_key].append(_element)
+        else:
+            _dict[_key] = [_element]
+
+    #
+    # Convert a dict into a string in JSON format.
+    #
+    def dict_to_json(self, _data, _pretty=False):
+        if _pretty:
+            return json.dumps(_data, sort_keys=True, indent=2)
+        else:
+            return json.dumps(_data)
+
+#
+# Main
+#
```

Comment:
considder adding an 
if __name__ == "__main__": 
statement. This prevents import side effects.